includes/class-sensei-messages.php

```diff
@@ -657,12 +657,12 @@ public function message_login() {
 		if ( is_single() && is_singular( $this->post_type )
 			|| is_post_type_archive( $this->post_type ) ) {
 
-			if ( isset( $my_courses_url ) ) {
+			$permalink = get_permalink();
 
-				wp_redirect( $my_courses_url, 303 );
+			if ( isset( $my_courses_url ) ) {
+				wp_safe_redirect( $my_courses_url . '?redirect_to=' . $permalink, 303 );
```

Comment:
Might be better to use [`add_query_arg()`](https://developer.wordpress.org/reference/functions/add_query_arg/) function.